src/components/composer/composer/prettifier_content.ts

```diff
@@ -0,0 +1,255 @@
+import {
+  AST,
+  leftOperandNeedsParenthesis,
+  rightOperandNeedsParenthesis,
+} from "../../../formulas/parser";
+
+// ---------------------------------------
+// Prettify possibility structure
+// ---------------------------------------
+
+// The `PrettifyPossibility` structure represents all possible ways to format an Abstract Syntax Tree (AST) into a human-readable string.
+// It includes formatting elements such as line breaks, indentation, and token grouping.
+//
+// Once created, this structure is used to determine the best formatting path based on the available width,
+// allowing dynamic selection of the most suitable layout for the given constraints.
+
+type PrettifyPossibility = string | ChooseBetween | Join | Nest | Line;
+type ChooseBetween = { type: "chooseBetween"; p1: PrettifyPossibility; p2: PrettifyPossibility };
+type Join = { type: "join"; rules: PrettifyPossibility[] };
+type Nest = { type: "nest"; indentLvl: number; p: PrettifyPossibility };
+type Line = { type: "line" };
```

Comment:
I would rather name  this 'insertLine' to match the action it conveys, just like you do with Nest and Join